### PR TITLE
fix(tweetRanking): :ambulance: fixed tweet ranking periods

### LINKED ISSUE #1251: Fix “since” date/hour from tweet ranking
The start date of the ranking is being initialized at 23:59, but it should start at 00h

image

### LINKED ISSUE #1252: Only the weekly ranking is working correctly
The cc sseraphini's tweets ranking should be running daily, but it is currently only being posted weekly, it is necessary to analyze to understand the reason.

### PR DESCRIPTION
**Pull request by pjonatansr**
closes #1251, closes #1252

### CODE REVIEW

File: apps/bot/src/date/getFirstDateOfMonth.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const getFirstDateOfMonth = (date: Date): Date => new Date(
```
Comment (fersilva16) [inline apps/bot/src/date/getFirstDateOfMonth.ts:1]: Prefer using a library

https://momentjscom.readthedocs.io/en/latest/moment/03-manipulating/03-start-of/

File: apps/bot/src/date/getDayBeforeDate.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const getDayBeforeDate = (date: Date): Date => new Date(
```
Comment (fersilva16) [inline apps/bot/src/date/getDayBeforeDate.ts:1]: Same

File: apps/bot/src/date/getWeekStartDate.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const getWeekStartDate = (date: Date): Date => new Date(
```
Comment (fersilva16) [inline apps/bot/src/date/getWeekStartDate.ts:1]: Same

File: apps/bot/src/date/getLastDayOfMonth.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const getLastDayOfMonth = (date: Date): number => new Date(
```
Comment (fersilva16) [inline apps/bot/src/date/getLastDayOfMonth.ts:1]: Same

**Review by fersilva16 — APPROVED** [review]

File: apps/bot/src/date/getFirstDateOfMonth.ts (hunk 1 above)
Comment (pjonatansr) [reply apps/bot/src/date/getFirstDateOfMonth.ts:1]: nice, I'll take a look

Comment (fersilva16) [reply apps/bot/src/date/getFirstDateOfMonth.ts:1]: I think we should use luxon btw, it's the evolution of moment

https://moment.github.io/luxon/#/?id=luxon

Comment (pjonatansr) [reply apps/bot/src/date/getFirstDateOfMonth.ts:1]: I'll merge this PR and open another to use luxon.

**Pull request merged**